templates/decays.hpp.in

```diff
@@ -0,0 +1,276 @@
+// ====================================================================
+// This file is part of FlexibleSUSY.
+//
+// FlexibleSUSY is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// FlexibleSUSY is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with FlexibleSUSY.  If not, see
+// <http://www.gnu.org/licenses/>.
+// ====================================================================
+
+// File generated at @DateAndTime@
+
+/**
+ * @file @ModelName@_decays.hpp
+ *
+ * @brief contains class for calculating particle decays
+ *
+ * This file was generated at @DateAndTime@ with FlexibleSUSY
+ * @FlexibleSUSYVersion@ (git commit: @FlexibleSUSYGitCommit@) and SARAH @SARAHVersion@ .
+ */
+
+#ifndef @ModelName@_DECAYS_H
+#define @ModelName@_DECAYS_H
+
+#include <boost/core/demangle.hpp>
+
+#include "@ModelName@_decay_table.hpp"
+#include "@ModelName@_mass_eigenstates.hpp"
+#include "@ModelName@_mass_eigenstates_decoupling_scheme.hpp"
+#include "@ModelName@_input_parameters.hpp"
+#include "cxx_qft/@ModelName@_qft.hpp"
+#include "@ModelName@_decay_amplitudes.hpp"
+#include "decays_loop_corrections.hpp"
+#include "decays_problems.hpp"
+#include "effective_couplings.hpp"
+#include "lowe.h"
+#include "wrappers.hpp"
+#include "error.hpp"
+
+namespace flexiblesusy {
+
+template <typename Field1, typename Field2>
+double
+final_state_symmetry_factor(typename @ModelName@_cxx_diagrams::field_indices<Field1>::type const& idx1,
+                            typename @ModelName@_cxx_diagrams::field_indices<Field2>::type const& idx2)
+{
+   if (std::is_same<Field1, Field2>::value) {
+      // TODO: we also compare if (potential) color indices are the same,
+      //       this might cause problems if someone accidentaly initializes them
+      //       differently
+      if (boost::range::equal(idx1, idx2)) {
+         return 0.5;
+      }
+   }
+   return 1.;
+}
+
+enum class HigherOrderSMCorrections {enable, disable};
```

Comment:
I'm wondering if this definition violates the ODR. For example, what happens when you include `*_decays.hpp` headers from multiple models in one file?

Reply:
Now I think this is definitely a problem, if multiple `<model>_decays.hpp` headers are included in the same translation unit. Please move the definition of this enum class to the `src/` directory, for example to `src/decays_loop_corrections.hpp`.

Reply:
And can this enum class be renamed to match the FS naming convention? I'd propose `SM_higher_order_corrections` or similar.

Reply:
You mean the case when we have multiple models at once? E.g. both `SM_decays.hpp` and `MSSM_decays.hpp`?

Reply:
Yes, exactly. In this case this type would be re-defined.

Reply:
In that case, shouldn't every model has it's own version? You might imagine that, for whatever reason, you'd like to use different precision in different models.

Reply:
That should of course be possible. But the definition of the *type* should be unique. The object (value) of the type may of course be different for different models.

Reply:
Done in f398cd7d07c448a023cc4840560f9e21c3ad40f6.

Reply:
Many thanks!